classes/module/Module.php

```diff
@@ -3432,33 +3436,47 @@ public function isAdminLegacyContext()
      */
     public function isSymfonyContext()
     {
-        return !$this->isAdminLegacyContext() && defined('_PS_ADMIN_DIR_');
+        $container = $this->getContainer();
+        if (null === $container) {
+            false;
+        }
+
+        return $container === SymfonyContainer::getInstance();
```

Comment:
Test for instance type instead? :thinking:

Reply:
It's actually tricky because all the containers (wether it's our custom built one or the one from kernel) all extend the same `Symfony\Component\DependencyInjection\Container`
But I found a way to trick this ^^ And introduced an internal `LegacyContainerInterface` that allows us to detect this without any loss of performance
I'm really open to suggestions about which namespace these new classes should be stored in though